dev/tools/controllerbuilder/template/controller/controller.go

```diff
@@ -226,32 +226,28 @@ func (a *{{.ProtoResource}}Adapter) Update(ctx context.Context, updateOp *direct
 
 
 	if len(paths) == 0 {
-		log.V(2).Info("no field needs update", "name", a.id.External)
-		status := &krm.{{.Kind}}Status{}
-		status.ObservedState = {{.Kind}}ObservedState_FromProto(mapCtx, a.actual)
-		if mapCtx.Err() != nil {
-			return mapCtx.Err()
-		}
-		return updateOp.UpdateStatus(ctx, status, nil)
+		log.V(2).Info("no field needs update", "name", a.id.String())
+		return nil
 	}
 	updateMask := &fieldmaskpb.FieldMask{
-		Paths: sets.List(paths)}
+		Paths: sets.List(paths),
+	}
 
 	// TODO(contributor): Complete the gcp "UPDATE" or "PATCH" request.
 	req := &{{.KCCService}}pb.Update{{.ProtoResource}}Request{
-		Name:       			a.id.External,
+		Name:       			a.id.String(),
 		UpdateMask:             updateMask,
 		{{.ProtoResource}}:     desiredPb,
 	}
 	op, err := a.gcpClient.Update{{.ProtoResource}}(ctx, req)
 	if err != nil {
-		return fmt.Errorf("updating {{.ProtoResource}} %s: %w", a.id.External, err)
+		return fmt.Errorf("updating {{.ProtoResource}} %s: %w", a.id.String(), err)
 	}
 	updated, err := op.Wait(ctx)
 	if err != nil {
-		return fmt.Errorf("{{.ProtoResource}} %s waiting update: %w", a.id.External, err)
+		return fmt.Errorf("{{.ProtoResource}} %s waiting update: %w", a.id.String(), err)
 	}
-	log.V(2).Info("successfully updated {{.ProtoResource}}", "name", a.id.External)
+	log.V(2).Info("successfully updated {{.ProtoResource}}", "name", a.id.String())
```

Comment:
nit: `a.id`. We don't need `String()`

Reply:
Done. I actually changed the wrong line. Now is fixed.

Reply:
I don't see any of these are fixed. Did you push to the right branch?